src/NHibernate/Engine/ActionQueue.cs

```diff
@@ -42,6 +43,7 @@ public partial class ActionQueue
 
 		private readonly AfterTransactionCompletionProcessQueue afterTransactionProcesses;
 		private readonly BeforeTransactionCompletionProcessQueue beforeTransactionProcesses;
+		private readonly HashSet<string> executedSpaces;
```

Comment:
@gliljas I'm still not sure why you need an additional collection for `executedSpaces` if you can use `AfterTransactionCompletionProcessQueue.querySpacesToInvalidate` (just make it internal) which holds exactly the same data?

Reply:
@hazzik I think that it's a bit wrong to to piggyback on a class that is not involved in the PreInvalidation. However, I have now moved them together, so that the spaces added to 
`AfterTransactionCompletionProcessQueue` comes from the same collection.

Reply:
I still don’t understand why you can not use the same collection